Code/Framework/AzToolsFramework/AzToolsFramework/UI/PropertyEditor/ComponentEditor.cpp

```diff
@@ -922,23 +922,26 @@ namespace AzToolsFramework
         return false;
     }
 
-    void ComponentEditor::EnteredComponentMode(const AZStd::vector<AZ::Uuid>& componentModeTypes)
+    ComponentEditor::ComponentEditorState ComponentEditor::EnteredComponentMode(const AZStd::vector<AZ::Uuid>& componentModeTypes)
     {
         // disable all component cards not matching the ComponentMode type
         if (AZStd::find(
             componentModeTypes.begin(),
             componentModeTypes.end(), m_componentType) == componentModeTypes.end())
         {
             SetWidgetInteractEnabled(this, false);
+            return ComponentEditor::ComponentEditorState::DidNotEnterComponentMode;
         }
         else
         {
             if (!componentModeTypes.empty())
             {
                 // only set the first item to be selected/highlighted
                 SetSelected(componentModeTypes.front() == m_componentType);
+                return ComponentEditor::ComponentEditorState::DidEnterComponentMode;
             }
         }
+        return ComponentEditor::ComponentEditorState::DidNotEnterComponentMode;
     }
```

Comment:
update: I don't think the enum naming here is right, as in both cases we do actually enter component mode afaik, it's just change how we visualize the Entity Inspector. I think here it's more like... `ComponentCardDisabled` and `ComponentCardSelected` - that's what's actually happening to the UI (and we only select the 'first' component card when we're entering two dependent Component Modes at once).